rpm_lockfile/__init__.py

```diff
@@ -163,22 +171,49 @@ def resolver(
             module_base.disable(module_disable)
             module_base.enable(module_enable)
 
-            # Mark packages to remove
-            for pkg in reinstall_packages:
+            if update:
+                # Updating subset of packages in existing lockfile. The
+                # packages and reinstallPackages sections are ignored in this
+                # code path.
+                for arch_data in lockfile["arches"]:
+                    if arch_data["arch"] != arch:
+                        # Not the correct arch, skip this.
+                        continue
+                    # Iterate over the packages in the lockfile ...
+                    for pkg in arch_data["packages"]:
+                        srpm = pkg["sourcerpm"].rsplit("-", 2)[0]
+                        # ... if the package belongs to an updated build ...
+                        if srpm in update:
+                            # ... install it just by name but require higher
+                            # version than we had before.
+                            logging.debug("Will update %s", pkg["name"])
+                            base.install(f"{pkg['name']} > {pkg['evr']}")
```

Comment:
The requirement for upgrade is not great. If there is no update available, we may want the transaction to rather be a no-op. However, it also changes what happens if `--update foo` is set but `foo` has `Requires: bar > ...` with a newer version than what is in the lockfile, and also `bar` is not set to be updated.

With the version constraint, the tool will error out with conflicting requests (`foo` wants newer `bar`, but we locked `bar` to older version).

Without the constraint, the transaction succeeds by DNF using the older version of `foo`, thus not updating anything and not reporting any error.